Automattic/woocommerce-payments
# Add platform checkout prototype

#### Changes proposed in this Pull Request









Adds platform checkout support prototype.

The PR introduces the `_wcpay_feature_platform_checkout` feature flag which, when enabled, will register a prototype of functionality related to platform checkout.

Note that in its current state the prototype is not safe for production use.

#### Testing instructions





1. Set the `_wcpay_feature_platform_checkout` option to (int) `1`.
2. The option to checkout via platform checkout should now appear in cart and checkout blocks.
3. Set the `_wcpay_feature_platform_checkout` option to (int) `0`.
4. The option to checkout via platform checkout should **not** appear in cart and checkout blocks.
5. Also verify that it is not possible to access REST endpoints without a valid nonce, the `wc_ajax_wcpay_init_platform_checkout` AJAX action, and that the `X-Wcpay-Platform-Checkout-User` header has no effect when the feature flag is disabled (and when enabled, all of these work).

-------------------

   - Does not apply - prototype behind a feature flag.
   - Does not apply - prototype behind a feature flag.

**Post merge**



   - Does not apply - prototype behind a feature flag.
